src/imgix-core-js.js

```diff
@@ -1,103 +1,126 @@
+  var md5 = _jsMd5;
+  var Base64 = _jsBase64;
+
+  var VERSION = '0.3.1';
+  var DEFAULTS = {
+    host: null,
+    useHTTPS: true,
+    includeLibraryParam: true
+  };
+
+  var ImgixClient = (function() {
+    function ImgixClient(opts) {
+      var key, val;
+
+      this.settings = {};
+
+      for (key in DEFAULTS) {
+        val = DEFAULTS[key];
+        this.settings[key] = val;
+      }
 
-  toString() {
-    let uri = new URI({
-      protocol: this.secure ? "https" : "http",
-      hostname: this.host,
-      path: this.path,
-      query: this._query()
-    });
-    return uri.toString();
-  }
+      for (key in opts) {
+        val = opts[key];
+        this.settings[key] = val;
+      }
 
-  toUrl(newParams) {
-    this.queryParams = Object.assign(this.queryParams, newParams);
-    return this;
-  }
+      if (!this.settings.host) {
+        throw new Error('ImgixClient must be passed a valid host');
+      }
 
-  _query() {
-    return URI.buildQuery(Object.assign(this._queryWithoutSignature(), this._signature()), false, false);
-  }
+      if (this.settings.includeLibraryParam) {
+        this.settings.libraryParam = "js-" + VERSION;
+      }
+
+      if (this.settings.useHTTPS) {
+        this.settings.urlPrefix = 'https://'
+      } else {
+        this.settings.urlPrefix = 'http://'
+      }
+    }
+
+    ImgixClient.prototype.buildURL = function(path, params) {
+      path = this._sanitizePath(path);
+
+      if (params == null) {
+        params = {};
+      }
 
-  _b64EncodedQuery() {
-    let query = this.queryParams;
-    let encodedQuery = {};
+      var queryParams = this._buildParams(params);
+      if (this.settings.secureURLToken != null) {
+        queryParams = this._signParams(path, queryParams);
+      }
 
-    for (let key in query) {
-      let val = query[key];
+      return this.settings.urlPrefix + this.settings.host + path + queryParams;
+    };
 
-      if (key.substr(-2) == '64') {
-        encodedQuery[key] = Base64.Base64.encodeURI(val);
+    ImgixClient.prototype._sanitizePath = function(path) {
+      if (path.indexOf('http') === 0) {
+        path = encodeURIComponent(path);
       } else {
-        encodedQuery[key] = val;
+        path = encodeURI(path);
       }
-    }
 
-    return encodedQuery;
-  }
+      if (path[0] !== '/') {
+        path = "/" + path;
+      }
 
-  _queryWithoutSignature() {
-    let query = this._b64EncodedQuery();
+      return path;
+    };
 
-    if (this.librarySignature && this.libraryVersion) {
-      query.ixlib = `${this.librarySignature}-${this.libraryVersion}`;
-    }
+    ImgixClient.prototype._buildParams = function(params) {
+      if (this.settings.libraryParam) {
+        params.ixlib = this.settings.libraryParam
+      }
 
-    return query;
-  }
+      var queryParams = [];
+      var key, val, encodedKey, encodedVal;
+      for (key in params) {
+        val = params[key];
+        encodedKey = encodeURIComponent(key);
```

Comment:
This is a new standard across all libraries. It's an unlikely case, but protects against anyone generating URLs from user input.